pkg/bootstrapper/bootstrapper.go

```diff
@@ -283,15 +317,19 @@ func (wmcb *winNodeBootstrapper) parseIgnitionFileContents(ignitionFileContents
 			continue
 		}
 
-		results := cloudProviderRegex.FindStringSubmatch(unit.Contents)
+		if unit.Contents == nil {
```

Comment:
AFAIK, it's good to have this guard but for Kubelet this will never be nil

Reply:
Contents is a pointer field in spec v3 -- I think we definitely want to guard against this here to avoid panics when it is nil for some unexpected reason.